polkadot/xcm/pallet-xcm-benchmarks/src/generic/benchmarking.rs

```diff
@@ -961,6 +961,33 @@ mod benchmarks {
 		Ok(())
 	}
 
+	#[benchmark]
+	fn barrier_check() -> Result<(), BenchmarkError> {
+		use sp_io::hashing::twox_128;
+		use xcm::latest::{prelude::Outcome, Error::Barrier};
+
+		let xcm = T::worst_case_for_not_passing_barrier().map_err(|_| BenchmarkError::Skip)?;
+		let mut executor = ExecuteXcmOf::<T>::new(Location::default(), XcmHash::default());
+
+		// Whitelist the hot read so it doesn't count towards PoV.
+		let record_xcm_key = [twox_128(b"PolkadotXcm"), twox_128(b"ShouldRecordXcm")].concat();
+		frame_benchmarking::benchmarking::add_to_whitelist(record_xcm_key.into());
```

Comment:
Shouldn't we mark it as `#[pallet::whitelist_storage]` so every benchmark benefits from this?